### ISSUE TITLE
GCC vs ARMCC deathmatch: compilation issues

### ISSUE THREAD

**Issue by PrzemekWirkus** [ARMmbed/utest#7]
When compiling with ARMCC 5 we have few issues related to how utest calls test cases.

With GCC:
```
{{timeout;15}}
{{host_test_name;default_auto}}
{{description;case control asynchronous test}}
{{test_id;MBED_OS}}
{{start}}
>>> Running 7 test cases...

>>> Running case #1: 'Control: Timeout (Failure)'...
>>> failure with reason 'Timed Out' during 'Case Handler'
>>> 'Control: Timeout (Failure)': 0 passed, 1 failed with reason 'Test Cases Failed'

>>> Running case #2: 'Control: Timeout (Success)'...
>>> 'Control: Timeout (Success)': 1 passed, 0 failed

>>> Running case #3: 'Control: Await'...
>>> 'Control: Await': 1 passed, 0 failed

>>> Running case #4: 'Control: RepeatAllOnTimeout'...
Running case handler for 1. time
>>> failure with reason 'Ignored: Timed Out' during 'Case Handler'
>>> 'Control: RepeatAllOnTimeout': 0 passed, 0 failed
```

With ARMCC:
```
{{timeout;15}}
{{ø{{timeout;15}}
{{host_test_name;default_auto}}
{{description;case control asynchronous test}}
{{test_id;MBED_OS}}
{{start}}
>>> Running 7 test cases...
>>> failure with reason 'Timed Out' during 'Case Handler'
>>> 'Control: Timeout (Failure)': 0 passed, 1 failed with reason 'Test Cases Failed'
>>> 'Control: Timeout (Success)': 1 passed, 0 failed

>>> Running case #4: 'Control: RepeatAllOnTimeout'...
Running case handler for 1. time
>>> failure with reason 'Ignored: Timed Out' during 'Case Handler'
>>> 'Control: RepeatAllOnTimeout': 0 passed, 0 failed
```

**Comment by PrzemekWirkus** [comment]
This is my compilation log from utest: [armccytbuild.txt](https://github.com/ARMmbed/utest/files/136228/armccytbuild.txt)

Other stuff:
```
$ yt ls -a
utest 1.9.1
|_ minar 1.0.4 (public registry)
| |_ compiler-polyfill >=1.1.0,<2.0.0
| |_ minar-platform 1.0.0 yotta_modules\minar-platform
| | |_ minar-platform-mbed 1.1.2 (public registry) yotta_modules\minar-platform-mbed
| | | |_ cmsis-core >=1.0.0,<2.0.0
| | | |_ mbed-hal 1.2.2 (public registry) yotta_modules\mbed-hal
| | | | |_ mbed-drivers *
| | | | \_ mbed-hal-freescale 1.0.0 yotta_modules\mbed-hal-freescale
| | | |   \_ mbed-hal-ksdk-mcu 1.0.8 (public registry) yotta_modules\mbed-hal-ksdk-mcu
| | | |     |_ uvisor-lib 1.0.12 (public registry) yotta_modules\uvisor-lib
| | | |     | \_ cmsis-core >=1.0.0,<2.0.0
| | | |     |_ mbed-hal *
| | | |     \_ mbed-hal-k64f 1.1.0 (public registry) yotta_modules\mbed-hal-k64f
| | | |       |_ mbed-hal *
| | | |       \_ mbed-hal-frdm-k64f 1.0.1 (public registry) yotta_modules\mbed-hal-frdm-k64f
| | | |         |_ mbed-hal *
| | | |         \_ mbed-hal-k64f >=1.1.0,<2.0.0
| | | \_ minar *
| | \_ cmsis-core 1.1.2 (public registry) yotta_modules\cmsis-core
| |   \_ cmsis-core-freescale 1.0.0 yotta_modules\cmsis-core-freescale
| |     \_ cmsis-core-k64f 1.0.0 yotta_modules\cmsis-core-k64f
| |       \_ cmsis-core *
| \_ core-util >=1.0.0,<2.0.0
|_ core-util 1.3.0 (public registry)
| |_ ualloc 1.0.3 (public registry) yotta_modules\ualloc
| | |_ dlmalloc 1.0.0 yotta_modules\dlmalloc
| | \_ core-util >=1.0.0,<2.0.0
| |_ cmsis-core >=1.0.0,<2.0.0
| \_ mbed-drivers >=0.12.0,<0.13.0
|_ compiler-polyfill 1.2.1 (public registry)
|_ unity 2.1.0 (public registry) (test dependency)
| \_ utest >=1.5.0,<2.0.0
\_ mbed-drivers 0.11.8 (public registry) (test dependency)
  |_ mbed-hal >=1.0.0,<2.0.0
  |_ cmsis-core >=1.0.0,<2.0.0
  |_ ualloc >=1.0.0,<2.0.0
  |_ minar >=1.0.0,<2.0.0
  |_ core-util >=1.0.0,<2.0.0
  \_ compiler-polyfill >=1.1.0,<2.0.0
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSFW-2121

**Comment by 0xc0170** [comment]
I have been having the same issue with armcc targets. armcc targets do not produce the same output (missing some information as pinpointed above).

**Comment by niklas-arm** [comment]
This has been fixed in #16. I believe the problem was C++11 member initialization for a static class, where the constructor was called too late.
